File: java/src/main/java/org/ikigaidigital/entity/TimeDepositEntity.java
package org.ikigaidigital.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.List;


@Entity
@Table(name = "TIMEDEPOSITS")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class TimeDepositEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private Integer id;
    @Column(name = "PLANTYPE")
    private String planType;
    @Column(name = "BALANCE")
    private Double balance;
    @Column(name = "DAYS")
    private int days;
    @OneToMany(mappedBy = "timeDeposit", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<WithdrawalEntity> withdrawals;

}

File: kotlin/src/main/kotlin/org/ikigaidigital/persistence/entity/TimeDepositEntity.kt
package org.ikigaidigital.persistence.entity

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table(name = "time_deposits")
class TimeDepositEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0,

    @Column(name = "plan_type", nullable = false)
    val planType: String,

    @Column(nullable = false)
    val days: Int,

    @Column(nullable = false, precision = 19, scale = 4)
    var balance: BigDecimal
)